mirrord-layer/src/connection.rs

```diff
@@ -106,10 +110,13 @@ pub(crate) async fn connect(config: &LayerConfig) -> impl AsyncWrite + AsyncRead
                 info!("No existing agent, spawning new one.");
                 let agent_port: u16 = rand::thread_rng().gen_range(30000..=65535);
                 info!("Using port `{agent_port:?}` for communication");
-                let pod_agent_name = match k8s_api.create_agent(agent_port).await {
-                    Ok(pod_name) => pod_name,
-                    Err(err) => handle_error(err),
-                };
+                let pod_agent_name = tokio::time::timeout(
+                    Duration::from_secs(config.agent.communication_timeout.unwrap_or(30) as u64),
+                    k8s_api.create_agent(agent_port),
+                )
+                .await
+                .unwrap_or(Err(LayerError::AgentReadyTimeout))
```

Comment:
I'm not sure how `unwrap_or` makes sense here or how it compiles.. I'd imagine using `map_err` here because you're converting one err type to another err type.

Reply:
Okay I do understand now. The return value is `Result<Result<T,E>,E>` still a bit weird. I'm okay with leaving as is.